bridges/snowbridge/pallets/system-v2/src/lib.rs

```diff
@@ -260,6 +231,20 @@ pub mod pallet {
 			T::OutboundQueue::deliver(ticket).map_err(|err| Error::<T>::Send(err))?;
 			Ok(())
 		}
+
+		/// Reanchor the `location` in context of ethereum
+		fn reanchor(location: &Location) -> Result<Location, Error<T>> {
+			location
+				.clone()
+				.reanchored(&T::EthereumLocation::get(), &T::UniversalLocation::get())
+				.map_err(|_| Error::<T>::LocationReanchorFailed)
+		}
+
+		fn location_to_agent_id(location: &Location) -> Result<AgentId, Error<T>> {
+			let reanchored_location = Self::reanchor(location)?;
+			AgentIdOf::convert_location(&reanchored_location)
+				.ok_or(Error::<T>::LocationConversionFailed)
+		}
```

Comment:
Agent is an ethereum-specific construct. Within this pallet, its just a hash of a reanchored location.

Otherwise you introduce confusion in the sense that reviewers expect that there is an `Agent` contract on Ethereum for this location.

```suggestion
		fn location_to_message_origin(location: &Location) -> Result<H256, Error<T>> {
			let reanchored_location = Self::reanchor(location)?;
			AgentIdOf::convert_location(&reanchored_location)
				.ok_or(Error::<T>::LocationConversionFailed)
		}
```

Reply:
Also see my comment above where I suggest that `T::AgentIdOf` (suggest to rename to `T::LocationHashOf`) also does reanchoring logic.

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/a7052509b6b5747fb1b9601c7e3a92f80b66cfff